test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs

```diff
@@ -0,0 +1,227 @@
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TestCases))]
+        public async Task LazilyLoad_WhenDoNotTrackChanges_ShouldNotCreateExtraRequest(string caseName, TestCaseHolder testCaseHolder)
+        {
+            using var store = GetDocumentStore();
+
+            var (expected, actual) = await AggressiveCacheOnLazilyLoadTest(store, testCaseHolder, AggressiveCacheMode.DoNotTrackChanges);
+            
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public async Task LazilyLoad_WhenTrackChangesAndChange_ShouldCreateExtraRequest()
+        {
+            using var store = GetDocumentStore();
+            
+            var (expected, actual) = await AggressiveCacheOnLazilyLoadTest(store, LazilyLoadFunc, AggressiveCacheMode.TrackChanges);
+
+            Assert.True(expected < actual, $"{expected} < {actual}");
+        }
+
+
+        [Fact]
+        public async Task LazilyLoad_WhenTrackChangesAndDoesntChange_ShouldNotCreateExtraRequest()
+        {
+            using var store = GetDocumentStore();
+
+            var (expected, actual) = await AggressiveCacheOnLazilyLoadTest(store, LazilyLoadFunc, AggressiveCacheMode.TrackChanges, false);
+
+            Assert.Equal(expected, actual);
+        }
+
+        private async Task<(long expected, long actual)> AggressiveCacheOnLazilyLoadTest(
+            IDocumentStore store, 
+            TestCaseHolder testCaseHolder,
+            AggressiveCacheMode doNotTrackChanges, 
+            bool createVersion2 = true)
+        {
+            return await AggressiveCacheOnLazilyLoadTest(store, testCaseHolder.LoadFunc, doNotTrackChanges, createVersion2);
+        }
+
+        private static async Task<(long expected, long actual)> AggressiveCacheOnLazilyLoadTest(
+            IDocumentStore store, 
+            Action<IDocumentSession, string> loadFunc, 
+            AggressiveCacheMode doNotTrackChanges, 
+            bool createVersion2 = true)
+        {
+            const string docId = "doc-1";
+
+            var requestExecutor = store.GetRequestExecutor();
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(new Doc(), docId);
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenSession())
+            using (session.Advanced.DocumentStore.AggressivelyCacheFor(TimeSpan.FromMinutes(5), AggressiveCacheMode.TrackChanges))
+            {
+                loadFunc(session, docId);
+            }
+
+            if (createVersion2)
+            {
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new Doc {Version = "2"}, docId);
+                    await session.SaveChangesAsync();
+                }
+            }
+
+            var expected = requestExecutor.NumberOfServerRequests;
```

Comment:
I think it will be better if you will take the value from session.Advanced.NumberOfRequests, here you might get a topology update so it is racy IMO